modules/packages/LinearAlgebra.chpl

```diff
@@ -232,27 +232,28 @@ of the matrix.
 */
 class ExpmPadeHelper {
   type matType;
-  var A : matType;
-  var A2 : matType;
-  var A4 : matType;
-  var A6 : matType;
-  var A8 : matType;
-  var A10 : matType;
-  var exactD4 : real;
-  var exactD6 : real;
-  var exactD8 : real;
-  var exactD10 : real;
-  var approxD4 : real;
-  var approxD6 : real;
-  var approxD8 : real;
-  var approxD10 : real;
-  var ident : matType;
-  var useExactOneNorm : bool;
+  var A: matType;
+  var A2: matType;
+  var A4: matType;
+  var A6: matType;
+  var A8: matType;
+  var A10: matType;
```

Comment:
Technically, these could be `var As: [2..10 by 2] matType`, right? I think that could make things a bit easier to read, even though it can hurt performance a tiny bit. I don't think the performance hit matters, that much, personally.

Note that, if you make that change, you should make similar changes to `exact*` and `approx*` below, and change the caching flag arrays to regular arrays instead of associative, as well.

Reply:
Reading the code, I think I want to make this change. It looks like you have a lot of repetition, and I feel like that repetition would go away, if you do this.

Something like

```chapel
proc comp_looseD(val: int) {
  ...
  ... this.approxD[val] ...
  ... this.compD(val) ...
}
```

should be able to get rid of the redundancy.

Reply:
(I also realize that this is nothing entirely, but it looks like this PR exacerbates an already existing issue)